### PR TITLE
apple_sdks: init with 11.1, 11.3, 12.1, 12.3, 13.1, and 13.3

### PR DESCRIPTION
**Pull request by ConnorBaker**
## What's new?

Adds a new directory for `apple-sdks` and refactors Xcode directory. Both use a “builder” and “releases” pattern where a nix configuration file stores information about each release and a separate nix file contains the function to build each release. This pattern has proven useful in the CUDA work I’ve done where we need to support multiple SDKs.

Default Apple SDK for `aarch64` has been bumped from 11.0 to 11.1.

Apple SDK versions 11.3, 12.1, 12.3, 13.1, and 13.3 are now available.

In addition, we now have a script to track Apple SDK releases (by parsing their `sucatalog`). The `gen-frameworks.py` script has been updated to accept command line arguments as well as support generating a `public.nix` file for each SDK release in the file generated by `gen-apple-sdk-releases.py`.

## Current problems?

- The 13.3 SDK is mean to be paired with Xcode 14.3, which is in turn backed by LLVM/Clang 15, so there are some compile errors when using old versions of LLVM/Clang.
- Cleanly overriding the compiler used by `stdenv` and ensuring linking against the correct `libc`/`libcxx`.
  - See https://github.com/NixOS/nixpkgs/issues/222717 and https://github.com/NixOS/nixpkgs/pull/229786
- There's a chance that jobs which have to rebuild perl for various stages of bootstrapping the `stdenv` compiler will fail if run in parallel. The specific error mentions miniperl. If encountered, limit the number of jobs to one until the bootstrapping is complete.

## Docs

- Please see `pkgs/os-specific/darwin/apple-sdks/README.md` for the beginnings of some documentation around how `frameworks` are generated.

## Miscellaneous notes

Notes on the SDK's `mkStdenv` in `default.nix`, courtesy of @reckenrode:

- While the minimum version of Darwin supported by the 13.3 SDK is 13.0, it’s possible to build against a newer SDK while targeting older platforms as long as you use an older deployment target and avoid (or dynamically check for) APIs specific to the newer SDK, so it's fine to set `darwinMinVersion = "10.12"`.
- Although the 13.3 SDK expects Xcode 14.3, we don't need to specify `xcodeVer = "14.3"`. Nor do we need to override the default Xcode set in `pkgs/os-specific/darwin/Xcode/default.nix` because it only serves to copy the Xcode into the store and there aren't any derivations in Nixpkgs which seem to use it.
- Because the 13.3 SDK is mean to be paired with Xcode 14.3, which is in turn backed by LLVM/Clang 15, there are some compile errors when using old versions of LLVM/Clang.
  - This is an open problem and requires further investigation
- We specify `darwinSdkVersion = "13.3"` so our SDK is chosen instead of the default when building `stdenv`.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/top-level/darwin-packages.nix [hunk 1]
```diff
@@ -33,8 +33,12 @@ makeScopeWithSplicing (generateSplicesForMkScope "darwin") (_: {}) (spliced: spl
   # macOS 11.0 SDK
   apple_sdk_11_0 = pkgs.callPackage ../os-specific/darwin/apple-sdk-11.0 { };
 
+  # macOS 13.3 SDK
+  apple_sdk_13_3 = pkgs.callPackage ../os-specific/darwin/apple-sdk-13 { };
+
   # Pick an SDK
-  apple_sdk = if stdenv.hostPlatform.isAarch64 then apple_sdk_11_0 else apple_sdk_10_12;
+  apple_sdk = apple_sdk_13_3;
```
Comment (wegank) [inline pkgs/top-level/darwin-packages.nix:40]: I think we should drop these changes to avoid a treewide rebuild on master.

File: pkgs/top-level/darwin-packages.nix [hunk 2]
```diff
@@ -108,9 +112,6 @@ impure-cmds // appleSourcePackages // chooseLibs // {
     stdenv = if stdenv.isDarwin then stdenv else pkgs.libcxxStdenv;
   };
 
-  # TODO: remove alias.
-  cf-private = self.apple_sdk.frameworks.CoreFoundation;
-
```
Comment (wegank) [inline pkgs/top-level/darwin-packages.nix:113]: Could you please do this in a separate PR? This could be merged much earlier than the SDK bump.

File: pkgs/os-specific/darwin/apple-sdk-13/default.nix [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+      };
+    in
+    if stdenv.isAarch64 then stdenv
+    else
+      (overrideCC stdenv cc).override {
+        targetPlatform = stdenv.targetPlatform // {
+          darwinMinVersion = "10.12";
+          darwinSdkVersion = "13.3";
+        };
+      };
```
Comment (reckenrode) [inline pkgs/os-specific/darwin/apple-sdk-13/default.nix:72]:
```suggestion
    (overrideCC stdenv cc).override {
      targetPlatform = stdenv.targetPlatform // {
        darwinMinVersion = "10.12";
        darwinSdkVersion = "13.3";
      };
    };
```

Assuming the default SDK will remain 11.0, this should be made unconditional. I’ve made the change locally to test, but I think MoltenVK is failing to build on aarch64-darwin because it would be getting an 11.0 SDK `stdenv` with 13.3 SDK frameworks.

Comment (wegank) [reply pkgs/os-specific/darwin/apple-sdk-13/default.nix:72]: > because it would be getting an 11.0 SDK `stdenv` with 13.3 SDK frameworks.

Perhaps something like `darwin.apple_sdk_13_3.callPackage`?

File: pkgs/top-level/darwin-packages.nix (hunk 1 above)
Comment (reckenrode) [reply pkgs/top-level/darwin-packages.nix:40]: Whether to bump the default SDK should be a separate PR. Going to 13.3 might be a little too aggressive for users who haven’t (or can’t) update to 13.3 yet.

File: pkgs/os-specific/darwin/apple-sdk-13/default.nix (hunk 3 above)
Comment (reckenrode) [reply pkgs/os-specific/darwin/apple-sdk-13/default.nix:72]: I missed the change in default SDK. I reverted that part, and the number of rebuilds are much lower. That should make troubleshooting the MoltenVK failure much easier.

Comment (reckenrode) [reply pkgs/os-specific/darwin/apple-sdk-13/default.nix:72]: It’s not picking up the 13.3 SDK `xcbuild`. 🤔

Comment (reckenrode) [reply pkgs/os-specific/darwin/apple-sdk-13/default.nix:72]: > > because it would be getting an 11.0 SDK `stdenv` with 13.3 SDK frameworks.
> 
> Perhaps something like `darwin.apple_sdk_13_3.callPackage`?

`mkStdEnv` is used to define the 13.3 SDK `stdenv`s provided by `darwin.apple_sdk_13_3.callPackage`, so it can’t use the 13.3 SDK `callPackage` without causing an infinite recursion.

The 11.0 SDK definition should also be made unconditional. When I added support for x86_64-darwin, I wanted to be very conservative about rebuilds, but it’s probably not necessary.

File: pkgs/top-level/darwin-packages.nix (hunk 1 above)
Comment (ConnorBaker) [reply pkgs/top-level/darwin-packages.nix]: Agreed, sorry that was mostly to simplify local testing

File: pkgs/top-level/darwin-packages.nix (hunk 2 above)
Comment (ConnorBaker) [reply pkgs/top-level/darwin-packages.nix:113]: See https://github.com/NixOS/nixpkgs/pull/229314

File: pkgs/os-specific/darwin/apple-sdk-13/default.nix (hunk 3 above)
Comment (ConnorBaker) [reply pkgs/os-specific/darwin/apple-sdk-13/default.nix]: Made those changes for the 13.3 SDK, but I can make them for the 11.0 SDK if you feel it wouldn't be out of scope.

Also added some general questions I had as TODOs:

https://github.com/NixOS/nixpkgs/pull/229210/commits/a80d48e2d5d137d001b0673e726081a4f4244417#diff-5b60d4457dd97671b469283ed86615b062fc1eeb634ba9331a8cd1e7a1d46366R67-R77

**Comment by fgaz** [comment]
>  ofborg bot requested review from fgaz

I wonder where did it get my username from. I don't see any change involving me and I don't have any darwin machine

**Comment by ConnorBaker** [comment]
Okay! Did some fairly major refactoring of the 13.3 SDK to genericize it and make it easier to sub in other versions. I've also added a `releases.nix` with different versions of the SDK. @reckenrode what do you think? Thoughts on the direction it took?

**Comment by reckenrode** [comment]
Should `releases.nix` be in the SDK or up a level? I like the idea of making the SDK more generic, so new versions can be added more easily, but how would this work with other SDKs? Did you have to do a lot of manual manipulate of the files from the generated ones?

**Comment by ConnorBaker** [comment]
> Should `releases.nix` be in the SDK or up a level? 

It should, but I'm hesitant to refactor *all* of the SDKs at the moment, so I'm trying to keep changes localized to the SDK for 13.

> I like the idea of making the SDK more generic, so new versions can be added more easily, but how would this work with other SDKs? Did you have to do a lot of manual manipulate of the files from the generated ones?

Thankfully, all of the 11.0+ releases seem to follow a similar pattern. I imagine each will have its own nix expression for public and private frameworks, as well as fixups, but that they'll have a lot in common. 

If we wanted to move away from checking in the auto-generated public dependencies, we could instead make use of `lib.optionals` to conditionally add different frameworks depending on the SDK version.

I just updated the PR description with a bit of documentation for what the `frameworks` dir is for currently. Let me know if that's at all coherent!

**Comment by ConnorBaker** [comment]
I want to note that I was also able to get rid of some recursion and switch over to using `callPackage` more consistently -- not sure what kind of breakages that will cause (if any at all) because I can't build things for the SDK right now.

**Comment by reckenrode** [comment]
> > Should `releases.nix` be in the SDK or up a level?
> 
> It should, but I'm hesitant to refactor _all_ of the SDKs at the moment, so I'm trying to keep changes localized to the SDK for 13.

Can it be at the top level without refactoring the other SDKs? That would allow them to be refactored later. Ideally, the SDKs should share as much of the build infrastructure to make adding new ones as painless as possible.

> > I like the idea of making the SDK more generic, so new versions can be added more easily, but how would this work with other SDKs? Did you have to do a lot of manual manipulate of the files from the generated ones?
> 
> Thankfully, all of the 11.0+ releases seem to follow a similar pattern. I imagine each will have its own nix expression for public and private frameworks, as well as fixups, but that they'll have a lot in common.
> 
> If we wanted to move away from checking in the auto-generated public dependencies, we could instead make use of `lib.optionals` to conditionally add different frameworks depending on the SDK version.

Would each dependency use `lib.optionals`? I’d fear that could get hard to read. Instead of using `lib.optionals`, could we adopt the override pattern? Each SDK would have an override function that receives the final set of dependencies and the unmodified set, then it returns a new set with whatever modifications it would need to make.

> I just updated the PR description with a bit of documentation for what the `frameworks` dir is for currently. Let me know if that's at all coherent!

Looks good!

One other thing that comes to mind is making sure the system ncurses is _excluded_ from the SDK. That’s not the case for the 11.0 SDK currently. It’s one of a few things that prevent Apple’s upstream cctools from building on aarch64-darwin. I grepped the current PR and don’t see it, so it appears it’s not excluded. Ideally, the 11.0 SDK would be updated/refactored and drop the system ncurses framework. Everything in nixpkgs should use the packaged ncurses.

There are a few private frameworks that also aren’t included. See https://github.com/NixOS/nixpkgs/issues/149692#issuecomment-1409604806. If those could also be included, it might be possible to build the upstream cctools and fix the linker crash.

**Comment by domenkozar** [comment]
cc @thefloweringash

**Comment by reckenrode** [comment]
I like the new organization. If this means adding new SDKs is a lot easier, that would be fantastic.

As discussed on Matrix, MoltenVK is well-behaved and can be built against the latest SDK. It uses dynamic feature detection to run on older systems. It supports a couple of macOS versions (10.11 and 10.12) for which it is impossible to build using those platforms’ SDKs.

**Comment by ConnorBaker** [comment]
I was able to remove the bundled `ncurses` from `Libsystem` and use `ncurses.dev` from Nixpkgs when building `cctools-apple`. Unfortunately, the problem is now `ld64`. I was able to fix the missing `corecrypto` header files by adding `Libsystem` to `buildInputs`, but now I see errors like the following:

```console
ld64> In file included from /nix/store/f8pjkgjvbgb1q5wsmpfyfb5v9s1l5354-libcxx-11.1.0-dev/include/c++/v1/__hash_table:18:
ld64> /nix/store/f8pjkgjvbgb1q5wsmpfyfb5v9s1l5354-libcxx-11.1.0-dev/include/c++/v1/cmath:321:9: error: no member named 'signbit' in the global namespace
ld64> using ::signbit;
ld64>       ~~^
ld64> /nix/store/f8pjkgjvbgb1q5wsmpfyfb5v9s1l5354-libcxx-11.1.0-dev/include/c++/v1/cmath:322:9: error: no member named 'fpclassify' in the global namespace
ld64> using ::fpclassify;
ld64>       ~~^
ld64> /nix/store/f8pjkgjvbgb1q5wsmpfyfb5v9s1l5354-libcxx-11.1.0-dev/include/c++/v1/cmath:323:9: error: no member named 'isfinite' in the global namespace
ld64> using ::isfinite;
ld64>       ~~^
ld64> /nix/store/f8pjkgjvbgb1q5wsmpfyfb5v9s1l5354-libcxx-11.1.0-dev/include/c++/v1/cmath:324:9: error: no member named 'isinf' in the global namespace
ld64> using ::isinf;
ld64>       ~~^
ld64> /nix/store/f8pjkgjvbgb1q5wsmpfyfb5v9s1l5354-libcxx-11.1.0-dev/include/c++/v1/cmath:325:9: error: no member named 'isnan' in the global namespace
ld64> using ::isnan;
ld64>       ~~^
ld64> /nix/store/f8pjkgjvbgb1q5wsmpfyfb5v9s1l5354-libcxx-11.1.0-dev/include/c++/v1/cmath:326:9: error: no member named 'isnormal' in the global namespace
ld64> using ::isnormal;
ld64>       ~~^
ld64> /nix/store/f8pjkgjvbgb1q5wsmpfyfb5v9s1l5354-libcxx-11.1.0-dev/include/c++/v1/cmath:327:7: error: no member named 'isgreater' in the global namespace; did you mean '::std::greater'?
ld64> using ::isgreater;
ld64>       ^~
```

Is it possible this is a problem with the default standard we're using to compile with? I know that there were changes involved in a compiler version or language standard at some point which removed common things from the global namespace (or how things were transitively included?). For example, I had a PR for a different project which failed to compile with GCC 12 because `array` was removed from a file's namespace and had to be explicitly included.

**Comment by reckenrode** [comment]
The issue is probably that Libsystem is conflicting with the C++ headers in nixpkgs. Instead of adding Libsystem to `buildInputs`, try manually symlinking `corecrypt` and add that path to `env.NIX_CFLAGS_COMPILE`. See [btop][1] (for `libproc.h`), [webkitgtk][2] (also for `libproc.h`), and [MoltenVK][2] (for the simd stuff) for examples of this workaround.

[1]: https://github.com/NixOS/nixpkgs/blob/7e7c514e829a18ca954516cd845744fe617bd9eb/pkgs/tools/system/btop/default.nix#L34
[2]: https://github.com/NixOS/nixpkgs/blob/7e7c514e829a18ca954516cd845744fe617bd9eb/pkgs/development/libraries/webkitgtk/default.nix#L173
[3]: https://github.com/NixOS/nixpkgs/blob/7e7c514e829a18ca954516cd845744fe617bd9eb/pkgs/os-specific/darwin/moltenvk/default.nix#L70

File: pkgs/os-specific/darwin/cctools/apple.nix [hunk 4]
```diff
@@ -1,118 +1,133 @@
...
+    };
+
+    patches = [
+      ./cctools-add-missing-vtool-libstuff-dep.patch
+    ];
+
+    postPatch = ''
+      for file in libstuff/writeout.c misc/libtool.c misc/lipo.c; do
+        substituteInPlace "$file" \
+          --replace '__builtin_available(macOS 10.12, *)' '0'
```
Comment (reckenrode) [inline pkgs/os-specific/darwin/cctools/apple.nix:35]: I wonder if we can drop this once the LLVM bump lands. It appears `__builtin_available` was added in Clang 12.

**Comment by reckenrode** [comment]
Thinking out loud a bit. I think this should be a follow up PR, but I’d also like to see the 10.{12, 13, 14, 15} SDKs added to this, and the source-based SDK defined as an overlay over the upstream frameworks.

* There are different attributes between source-based and non-source SDKs. Harmonizing them would get rid of some of the friction when having to work with both SDKs (e.g., different sdk / MacOSX-SDK attributes).
* An overlay should facilitate an incremental approach to “ensourcifying” the SDK. The current approach seems to be all or nothing, which makes the barrier to bumping too high.
* Removal of `__propagatedImpureHostDeps`. The newer SDKs don’t seem to have any `__propagatedImpureHostDeps`, which I assume is because they use `.tbd` files instead of linking against the frameworks.

The last point is a pretty big assumption about whether we can put all the SDKs in this framework. Having them all work and build roughly the same way would go a long way towards making it easier to work with Darwin, bump the SDKs, and add more source-based components.

**Comment by ConnorBaker** [comment]
Rebased and squashed!

**Comment by ConnorBaker** [comment]
Made a first pass at generating information for OSS distributions using https://github.com/NixOS/nixpkgs/blob/7713b3235abf54d8b5cdc25576df3917134cc76f/pkgs/os-specific/darwin/apple-source-releases/generate-distribution-releases.py.

Haven't tried building any of the actual distributions with that, but should be doable using the releases/generic builder pattern.

**Comment by ConnorBaker** [comment]
> This PR already adds 50k lines of vendored json and so far we lived a long time without every minor version.

That’s my bad; the OSS stuff should really be done in a separate PR. When I get a chance I’ll revert those changes. That should drop it down by a lot.

**Comment by ConnorBaker** [comment]
Rebased and reverted the commit dealing with OSS stuff.

File: pkgs/os-specific/darwin/apple-sdks/frameworks/11.3.0/fixups.nix [hunk 5]
```diff
@@ -0,0 +1,102 @@
+# TODO(@connorbaker): Copied from 11.0 -- should be regenerated.
```
Comment (Atemu) [inline pkgs/os-specific/darwin/apple-sdks/frameworks/11.3.0/fixups.nix:1]: I'd prefer if we had a single declaration which handles version differences internally.

DRY.

File: pkgs/tools/misc/fclones/default.nix [hunk 6]
```diff
@@ -19,7 +19,7 @@ rustPlatform.buildRustPackage rec {
   cargoHash = "sha256-C7DKwEMYdypfItflMOL7rjbAdXDRsXDNoPlc9j6aBRA=";
 
   buildInputs = lib.optionals stdenv.isDarwin [
-    darwin.apple_sdk_11_0.frameworks.AppKit
+    darwin.apple_sdk_11_1.frameworks.AppKit
```
Comment (Atemu) [inline pkgs/tools/misc/fclones/default.nix:22]: Perhaps add a `apple_sdk_11` or even just `apple_sdk_new`. What the vast majority of these packages actually want is a non-ancient SDK, not 11.0 or 11.1 specifically.

**Review by Atemu — COMMENTED** [review]
Could you split the PR into sensible commits?

One for adding new SDKs, one for bumping, one for other fixes etc.

File: pkgs/tools/misc/fclones/default.nix (hunk 6 above)
Comment (wegank) [reply pkgs/tools/misc/fclones/default.nix:22]: Proposed `darwin.apple_sdk_11` in #234763, but lost interest as there were no feedbacks...

**Comment by reckenrode** [comment]
@ConnorBaker fyi, https://github.com/NixOS/nixpkgs/pull/241692 has been merged into staging and should hopefully make 23.11.

**Comment by reckenrode** [comment]
@ConnorBaker #234710 is complete. The Darwin stdenv update has been merged into master.

**Comment by heimalne** [comment]
Any updates on the >= 12.x versions? This seems to be the new target architecture for many packages in the Python ecosystem and if i remember correctly the first official version to support Arm64 macs.

**Comment by reckenrode** [comment]
> Any updates on the >= 12.x versions?

I will be picking this up after I get the cctools and ld64 updates successfully building the Darwin blockers channel and take a look at the Darwin cross-compilation issue after those and the recent LLVM changes. The plan is to land an SDK refactor into a common pattern first then add the missing SDKs (10.13 through 14.x) with as many source-based components as possible without regressing compatibility.

Once newer SDKs are available, packages that need one can use `overrideSDK` to change the SDK version used to build them.

> This seems to be the new target architecture for many packages in the Python ecosystem and if i remember correctly the first official version to support Arm64 macs.

11.0 was the first release of macOS to support Apple Silicon. There are no plans to change the default SDK on aarch64-darwin, but x86_64-darwin will be updated to 10.13 or 10.14 (but most likely 10.13).

**Comment by gador** [comment]
@reckenrode did I read that correctly that you continue on this PR?
If so, that would be awesome! An update to the apple sdk system would be much appreciated! If you need some testing, let me know!

**Comment by reckenrode** [comment]
@gador That’s right (more or less). The first iteration will be just the current SDKs in the common pattern. I’d like to do it without causing rebuilds, but I’m doubtful it’s possible. It may be a continuation, or it may just incorporate some of the approach. I’ll post an update to the Darwin updates news thread on Discourse once I have something to test.

**Comment by grahamc** [comment]
Hey @reckenrode, I didn't mean to step on any toes with that flake. It was for an experiment, but I never managed to make it actually work. It'd definitely be better being part of nixpkgs, and we don't have any grand plans to maintain an apple sdk flake long term. Just trying to get out of a jam.

**Comment by reckenrode** [comment]
> Hey @reckenrode, I didn't mean to step on any toes with that flake. It was for an experiment, but I never managed to make it actually work. It'd definitely be better being part of nixpkgs, and we don't have any grand plans to maintain an apple sdk flake long term. Just trying to get out of a jam.

It’s not about whether it’s a flake or an alternative. It’s the lack of communication and roadmap, especially in light of recent controversies. Is this going to be the solution? Is DetSys going to drive that? What’s going on?

**Comment by grahamc** [comment]
There is no roadmap, and we're definitely not the people to decide if this is the solution, and we're not driving it, lol. Like I said: Just trying to get out of a jam where we needed a newer SDK for a Swift program, but couldn't wait for a future merge, and it wasn't working as-is in the PR.

**Comment by reckenrode** [comment]
@grahamc Thanks for the clarification. I think if I had known the context when I saw the repo, I’d have reacted better. It’s understandable that you needed a solution when the soonest anything could possibly land is after the 24.05 release. The current SDK situation is definitely not great.

**Comment by niklaskorz** [comment]
The zed-editor package will require the 12.3 SDKs to fix darwin because of [ScreenCaptureKit](https://developer.apple.com/documentation/screencapturekit/). I'll open a draft PR for the zed-editor fixes this week, if there's any way I can help please let me know. 😄

**Comment by aviallon** [comment]
Closed by https://github.com/NixOS/nixpkgs/pull/346043
